sdcm/nemesis.py

```diff
@@ -1072,6 +1073,79 @@ def disrupt_ldap_connection_toggle(self):
         ContainerManager.unpause_container(self.tester.localhost, 'ldap')
         self.log.info('finished with nemesis')
 
+    def disrupt_ldap_role_add_remove_permission(self):
+        """
+        Scenario:
+        1. Create new user in Scylla
+        2. Create new role in Ldap containing the new user as a member.
+        3. Verify this new user is unauthorised on resources (create a keyspace).
+        4. Create a new super-user role in Scylla.
+        5. Create a new super-user role in Ldap, with the new user as a member.
+        6. Verify this new user is authorised on resources.
+        7. Modify Ldap role to remove membership of new user.
+        8. Verify the new user is unauthorised on resources (create a table).
+        9. Do clean-up: delete Ldap Role, Scylla roles and keyspace/tables.
+
+        """
+        if not self.cluster.params.get('use_ldap_authorization'):
+            raise UnsupportedNemesis('Cluster is not configured to run with LDAP authorization, hence skipping')
+        if not self.target_node.is_enterprise:
+            raise UnsupportedNemesis('Cluster is not enterprise. LDAP is supported only for enterprise. Skipping')
+        node = self.cluster.nodes[0]
+        superuser_role = 'superuser_role'
+        scylla_qa2 = 'scylla_qa2'
+        self.log.debug("Create new user in Scylla")
+        self.tester.create_role_in_scylla(node=node, role_name=scylla_qa2, is_superuser=False,
+                                          is_login=True)
+        with pytest.raises(Unauthorized, match="has no CREATE permission"):
+            with self.cluster.cql_connection_patient(node=node, user=scylla_qa2, password=LDAP_PASSWORD) as session:
+                session.execute(
+                    """ CREATE KEYSPACE IF NOT EXISTS customer WITH replication = {'class': 'SimpleStrategy',
+                    'replication_factor': 1} """)
+
+        self.log.debug("Create a new super-user role in Scylla")
+        self.tester.create_role_in_scylla(node=node, role_name=superuser_role, is_superuser=True,
+                                          is_login=False)
+
+        self.log.debug("Create a new super-user role in Ldap, associated with the new user")
+        if self.cluster.params.get('prepare_saslauthd'):
+            self.tester.add_user_in_ldap(username=scylla_qa2)
+        self.tester.create_role_in_ldap(ldap_role_name=superuser_role, unique_members=[scylla_qa2, LDAP_USERS[1]])
+        self.cluster.wait_for_schema_agreement()
```

Comment:
Alternatively to this 'wait' i considered having something like `wait_ldap_role_update()` but might be too much overhead.

Reply:
Will possibly be resolved by a waiter method like:
```
@retrying(n=10, sleep_time=3, message='Waiting for user permissions update', allowed_exceptions=(AssertionError,))
def wait_for_user_permission_update(self, are_permissions_expected: bool, node: BaseNode, username):
    """
    Checks for updated output of user permissions.
    Example command: list ALL PERMISSIONS OF 'scylla_qa2'
    Example output:
         role       | username   | resource                  | permission
        ------------+------------+---------------------------+------------
         scylla_qa2 | scylla_qa2 |       <keyspace customer> |      ALTER
    """
    self.log.debug("Waiting user %s permissions change in Scylla DB after an LDAP Role update ", username)
# TODO: implement
```

Reply:
fixed